packages/epo-react-lib/vite.config.ts

```diff
@@ -2,29 +2,167 @@ import { defineConfig } from "vite";
+        Video: resolve(__dirname, "src/atomic/Video/Video.tsx"),
+        SimpleTable: resolve(
+          __dirname,
+          "src/content-blocks/SimpleTable/SimpleTable.tsx"
+        ),
+        ComplexTable: resolve(
+          __dirname,
+          "src/content-blocks/ComplexTable/ComplexTable.tsx"
+        ),
+        Error: resolve(__dirname, "src/form/Error/Error.tsx"),
+        FormButtons: resolve(__dirname, "src/form/FormButtons/FormButtons.tsx"),
+        FormField: resolve(__dirname, "src/form/FormField/FormField.tsx"),
+        HorizontalSlider: resolve(
+          __dirname,
+          "src/form/HorizontalSlider/HorizontalSlider.tsx"
+        ),
+        Input: resolve(__dirname, "src/form/Input/Input.tsx"),
+        Password: resolve(
+          __dirname,
+          "src/form/Input/patterns/Password/Password.tsx"
+        ),
+        Select: resolve(__dirname, "src/form/Select/Select.tsx"),
+        SelectListbox: resolve(
+          __dirname,
+          "src/form/SelectListbox/SelectListbox.tsx"
+        ),
+        Switch: resolve(__dirname, "src/form/Switch/Switch.tsx"),
+        BasicModal: resolve(__dirname, "src/layout/BasicModal/BasicModal.tsx"),
+        CarouselLayout: resolve(__dirname, "src/layout/Carousel/Carousel.tsx"),
+        Columns: resolve(__dirname, "src/layout/Columns/Columns.tsx"),
+        Container: resolve(__dirname, "src/layout/Container/Container.tsx"),
+        Grid: resolve(__dirname, "src/layout/Grid/Grid.tsx"),
+        MasonryGrid: resolve(
+          __dirname,
+          "src/layout/MasonryGrid/MasonryGrid.tsx"
+        ),
+        SlideoutMenu: resolve(__dirname, "src/layout/SlideoutMenu"),
+        IconComposer: resolve(
+          __dirname,
+          "src/svg/IconComposer/IconComposer.tsx"
+        ),
+        icons: resolve(__dirname, "src/svg/icons"),
+      },
+      formats: [defaultFormat, "cjs"],
+      fileName: (format, name) =>
+        `${name}.${format === defaultFormat ? "js" : format}`,
     },
     rollupOptions: {
+      plugins: [
+        banner2((chunk) => {
+          const useClient = `"use client"\n\n`;
+          const hooks = ["useState", "useEffect", "useRef", "useContext"];
+          const { importedBindings } = chunk;
+
+          if (Object.keys(importedBindings).length > 0) {
+            if (
+              importedBindings["styled-components"] &&
+              importedBindings["styled-components"].length > 0
+            ) {
+              return useClient;
+            }
+
+            if (
+              importedBindings["react"] &&
+              importedBindings["react"].filter((value) => hooks.includes(value))
+            ) {
+              return useClient;
+            }
+
+            if (
+              importedBindings["react-i18next"] &&
+              importedBindings["react-i18next"].includes("useTranslation")
+            ) {
+              return useClient;
+            }
+          }
+        }),
+      ],
```

Comment:
this banner attempts to insert `use client` in some broad areas where we know the components must be client components, i.e. anywhere `styled-components` is imported, `useTranslation`, or the major React hooks. This can be tightened up later and hopefully Vite will improve support for leaving directives untouched